x-pack/plugins/maps/public/shared/layers/styles/vector_style.js

```diff
@@ -215,8 +238,49 @@ export class VectorStyle extends AbstractStyle {
     return type === VectorStyle.STYLE_TYPE.DYNAMIC && options.field && options.field.name;
   }
 
-  _getIsPointsOnly = () => {
-    return _.get(this._descriptor, '__styleMeta.isPointsOnly', false);
+  getIsPointsOnly = async () => {
+    const supportedFeatures = await this._source.getSupportedFeatures();
+
+    if (supportedFeatures.length === 1) {
+      return supportedFeatures[0] === VECTOR_FEATURE_TYPES.POINT;
+    }
+
+    if (!this._descriptor.__styleMeta) {
+      return false;
+    }
+
+    const { hasPoints, hasLines, hasPolygons } = this._descriptor.__styleMeta;
+    return hasPoints && !hasLines && !hasPolygons;
+  }
+
+  getIsLinesOnly = async () => {
```

Comment:
consider creating helpermethod for these 3, they are identical

`_checkIfOnlyFeatureType(featureType) {...}`

This reuse will also be easier if we do the refactoring of the stylesMeta descriptor to use actual VECTOR_FEATURE_TYPE constants (see other comment).